crates/hyperqueue/src/server/event/journal/stream.rs

```diff
@@ -83,11 +93,28 @@ async fn streaming_process(
                            And while this read is performed, we cannot allow modification of the file,
                            so the writing to the file has to be paused anyway */
                         writer.flush()?;
-                        let mut reader = JournalReader::open(log_path)?;
+                        let mut reader = JournalReader::open(journal_path)?;
                         for event in &mut reader {
                             tx.send(event?).unwrap()
                         }
-                    }
+                    },
+                    Some(EventStreamMessage::PruneJournal { live_jobs, live_workers, callback })  => {
+                        writer.flush()?;
+                        let mut tmp_path: OsString = journal_path.into();
```

Comment:
It would be safer to create the journal file in a temporary directory, to avoid potential race conditions. Although we don't know where /tmp is located, so maybe try to atomically create a unique file in the directory of the journal?